crates/shared-crypto/src/intent.rs

```diff
@@ -44,10 +41,7 @@ pub enum AppId {
 impl TryFrom<u8> for AppId {
     type Error = eyre::Report;
     fn try_from(value: u8) -> Result<Self, Self::Error> {
-        match value {
-            0 => Ok(Self::Sui),
-            _ => Err(eyre!("Invalid AppId")),
-        }
+        bcs::from_bytes(&[value]).map_err(|_| eyre!("Invalid AppId"))
```

Comment:
Two questions:
a) do we use Narwhal AppId anywhere in our codebase? @benr-ml I only see one change related at the end of this PR, is this the sole place?
b) @benr-ml did you did a thorough check/search that we covered everything?

Reply:
Yes, there is only one place with NW. You can see that in the PR (since previously you had to call `with_app_id` that doesn't exist now)

Reply:
Ok I guess both creating NW headers and reading headers from senders are reusing the same function, right? I'd typically expect at least two places (write and read) unless we indeed reuse.